src/main/java/calculator/DelimiterService.java

```diff
@@ -0,0 +1,49 @@
+package calculator;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+
+public class DelimiterService {
+
+    private static final String COMMA = ",";
+    private static final String COLON = ":";
+
+    // 구분자와 숫자를 한 번에 처리하는 메서드
+    public List<String> processInputAndExtractNumbers(String input) {
+        String[] lines = input.split("\\\\n");
+        String delimiterPattern = getDefaultDelimiterPattern();
+
+        if (lines.length == 2) {
+            delimiterPattern = getCustomDelimiterPattern(lines[0]);
+            return Arrays.asList(lines[1].split(delimiterPattern));
+        }
+
+        return Arrays.asList(lines[0].split(delimiterPattern));
+
+    }
+
+    // 커스텀 구분자를 추출하고 정규식 패턴으로 반환하는 메서드
+    private String getCustomDelimiterPattern(String customDelimiterLine) {
+        // 커스텀 구분자 형식 체크
+        if (!customDelimiterLine.startsWith("//") || customDelimiterLine.length() != 3) {
+            throw new IllegalArgumentException("잘못된 구분자 형식입니다.");
+        }
```

Comment:
```suggestion
        if (isCustomDelimiter()) {
            throw new IllegalArgumentException("잘못된 구분자 형식입니다.");
        }
```
이런식으로 추상화하면 더 좋을 것 같아요 ! 그리고 매직넘버나 매직스트링을 상수처리하면 좋을 것 같아요 ! "//" 같은 경우는 상수로 관리한다면 추후 유지보수에도 좋다고 생각합니다 !

Reply:
네 민혁님 코드 보면서 느꼈습니다. 매직넘버, 스트링 관리를 조금 더 신경 써야 할 것 같아요.